### PR TITLE
Add place detail page

### PR DESCRIPTION
**Pull request by helenrivne**
Turned off page animation (will refactor animation), changed template, add preload images, remove code style problems

### CODE REVIEW

File: client/components/place/place-detail/place-detail.component.js [hunk 1]
```diff
@@ -1,26 +1,33 @@
...
-        center: [50.6234, 26.2189],
-        zoom: 14
+        center: constants.mapCenter,
+        zoom: constants.defaultZoom-6
       });
-      Streets = L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+      this.map.touchZoom.disable();
+      this.map.dragging.disable();
+      this.map.scrollWheelZoom.disable();
+      var layerStreet = L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.component.js]: Use `layers` from [**mapMarkingTypes.factory.js**](https://github.com/Social-projects-Rivne/GreenTourism/blob/90b360dffbd69adb27f27eddcfc549de248dc713/client/components/place/mapMarkingTypes.factory.js#L46)

File: client/components/place/place-detail/place-detail.component.js [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
       });
-      this.map.addLayer(Streets);
-      this.location = this.place.getLocation();
-      this.marker = L.marker(this.location).addTo(this.map);
-
-      this.marker.bindPopup('<div><h3>' + this.place.name +
-        '</h3><a><img class="marker-image" src="assets/' + this.place.photo[0] +
-        '" /></a><br />').openPopup();
+      this.map.addLayer(layerStreet);
+      this.marker = L.marker(L.latLng(this.place.location.coordinates[1], this.place.location.coordinates[0])).addTo(this.map);
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.component.js]: Make this line smaller than 80 characters

File: client/components/place/place-detail/place-detail.component.js [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
+      this.marker.bindPopup('<div class="popup"><h3>' + this.place.name +
+        '</h3><a><img class="marker-image center-block" src="' + this.place.photos[0] + '" /></a><br />').openPopup();
+      var deltaheight = 0.1;
+      this.map.setView([this.place.location.coordinates[1]+ deltaheight, this.place.location.coordinates[0]]); //0.1- for responcive design- show info on mobile
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.component.js]: Make this line smaller than 80 characters and remove comment from there (if you want comment to stay, move it to line 25, where `deltaheight` variable is declared)

File: client/components/place/place-detail/place-detail.template.html [hunk 4]
```diff
@@ -1,73 +1,74 @@
...
-        <br>
-        <div class="pull-left">
-          <img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="img-circle  profile"
-          />
-        </div>
-        <p class="clearfix">Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1
-          Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1</p>
+        <!--
+          <a ng-repeat="photo in $ctrl.place.photos" ng-if="$index>=1" class="fancybox" rel="gallery1"
+             href="{{photo}}" title="{{$ctrl.place.name}}">
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Use `ng-href`

File: client/components/place/place-detail/place-detail.template.html [hunk 5]
```diff
@@ -1,73 +1,74 @@
...
+            <img src="assets/{{photo}}" alt="{{$ctrl.place.name}}"/>
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Remove `assets/` and use `ng-src`

File: client/components/place/place-detail/place-detail.template.html [hunk 6]
```diff
@@ -1,73 +1,74 @@
...
-        </form>
-        <br>
-        <br>
-        <div class="pull-left">
-          <img src="assets/img/user-profile/anderson.jpeg" alt="Aniston" class="img-circle  profile"
-          />
-        </div>
-        <p class="clearfix">Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1
-          Comment1 Comment1 Comment1 Comment1 Comment1 Comment1 Comment1</p>
+        <!--
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: Remove this comment begging line

File: client/components/place/place-detail/place-detail.template.html [hunk 7]
```diff
@@ -1,73 +1,74 @@
...
+          <a ng-repeat="photo in $ctrl.place.photos" ng-if="$index>=1" class="fancybox" rel="gallery1"
+             href="{{photo}}" title="{{$ctrl.place.name}}">
+            <img src="assets/{{photo}}" alt="{{$ctrl.place.name}}"/>
+        </a>
+        <!-- to delete- it is for demonstrate without data about other photo in DB -->
```
Comment (yurykozhenov) [inline client/components/place/place-detail/place-detail.template.html]: If it's _to delete_ then delete it

File: client/components/place/placesOnMap.factory.js [hunk 8]
```diff
@@ -41,17 +41,21 @@ angular.module('mapModule')
...
         mainGroup.checkIn(groups[input]);
         groups[input].addTo(map);
       } else {
         places.forEach(function(place) {
           marker(place.location.coordinates[0], place.location.coordinates[1], types[place.type].icon)
             .addTo(groups[place.type])
-            .bindPopup('<div class=\'popup  center-block\'><h3>' + place.name + '</h3><a><img class=\'marker-image\' src=\'assets/' + place.photos[0] + '\' \/></a>' +
-              '<br /><br /><button type=\'button\' class=\'btn btn-default btn-md center-block\'> <a href=\'#!/places/' + place._id + '\'>Details >></a> </button></div>', {autoPan: false});
+            .bindPopup('<div class=\'popup\'><h3>' + place.name + '</h3>' +
+              '<a href=\'#!/places/' + place._id + '\'><img class=\'marker-image  center-block\' src=\''+place.photos[0]+'\' \/></a>' +
```
Comment (yurykozhenov) [inline client/components/place/placesOnMap.factory.js]: Make this line smaller than 80 characters

File: client/components/place/placesOnMap.factory.js [hunk 9]
```diff
@@ -41,17 +41,21 @@ angular.module('mapModule')
         places.forEach(function(place) {
           marker(place.location.coordinates[0], place.location.coordinates[1], types[input].icon)
             .addTo(groups[input])
-            .bindPopup('<div class=\'popup  center-block\'><h3>' + place.name + '</h3><a><img class=\'marker-image\' src=\'assets/' + place.photos[0] + '\' \/></a>' +
-              '<br /><br /><button type=\'button\' class=\'btn btn-default btn-md center-block\'> <a href=\'#!/places/' + place._id + '\'>Details >></a> </button></div>', {autoPan: false});
+            .bindPopup('<div class=\'popup\'><h3>' + place.name + '</h3>' +
+              '<a href=\'#!/places/' + place._id + '\'><img class=\'marker-image  center-block\' src=\''+place.photos[0]+'\' \/></a>' +
```
Comment (yurykozhenov) [inline client/components/place/placesOnMap.factory.js]: Make this line smaller than 80 characters

File: client/app.routes.js [hunk 10]
```diff
@@ -78,3 +78,6 @@ angular.module('greenTourism').config(['$routeProvider', '$locationProvider',
         templateUrl: 'shared/errors/404.html'
       });
   }]);
+angular.module('greenTourism').controller('placeDetailCtrl', function($scope) {
+  $scope.pageClass = 'page-detail';
+});
```
Comment (max-yushchuk) [inline client/app.routes.js]: Add line break at the end of file

File: client/assets/less/pages/place-detail.less [hunk 11]
```diff
@@ -0,0 +1,105 @@
+.page-detail {
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less:1]: I can not find this class in place-detail.template. Where is it use? And is it important?

Comment (yurykozhenov) [inline client/assets/less/pages/place-detail.less:1]: It's for animation. Used inside `index.ejs`

File: client/assets/less/pages/place-detail.less [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+.page-detail.ng-leave {
+  animation: slideInOut 0.5s both ease-in;
+  z-index: 9999;
+}
+
+div.place {
+  margin: 0;
+  .vertical-align {
+
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less:61]: Remove line break

File: client/assets/less/pages/place-detail.less [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    .gallery {
+      text-align: center;
+      a.fancybox {
+        max-height: 100px;
+        overflow: hidden;
+        display: inline-block;
+        width: 32%;
+      }
+      a img {
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less:103]: Put img into .fancybox. It's much better for reading code. Like:

``` css
.fancybox {
  img {
  }
}
```

File: client/components/place/place-detail/place-detail.component.js [hunk 14]
```diff
@@ -1,26 +1,33 @@
...
+angular.module('placeDetail', ['comment'])
   .component('placeDetail', {
     templateUrl: 'components/place/place-detail/place-detail.template.html',
     bindings: {
       place: '<'
     },
-    controller: ['mapFactory', function placeDetailCtrl(mapFactory) {
+    controller: function placeDetailCtrl($scope, constants) {
+      angular.element(document).ready(function() {
+        $('.fancybox').fancybox();
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.component.js]: Change $ on angular.element

File: client/components/place/place-detail/place-detail.component.js [hunk 15]
```diff
@@ -1,26 +1,33 @@
...
+      });
       this.map = L.map('map1', {
-        center: [50.6234, 26.2189],
-        zoom: 14
+        center: constants.mapCenter,
+        zoom: constants.defaultZoom-6
       });
-      Streets = L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+      this.map.touchZoom.disable();
```
Comment (max-yushchuk) [inline client/components/place/place-detail/place-detail.component.js]: Use in map:

```
touchZoom: false,
dragging: false,
scrollWheelZoom: false
```

instead this.map.touchZoom.disable(); etc.

File: client/assets/img/places/Новый текстовый документ.txt [hunk 16]
```diff
@@ -0,0 +1,19 @@
+http://rvavto.com.ua/images/greentourism/vnutri_tarakonovskogo_forta.jpg
```
Comment (yurykozhenov) [inline client/assets/img/places/Новый текстовый документ.txt]: Remove this file from repository

File: client/assets/less/pages/place-detail.less [hunk 17]
```diff
@@ -0,0 +1,202 @@
+.page-detail {
+  bottom: 0;
+  width: 100%;
+  top: 0;
+  background-color: #FFF;
+  .fa-spin-custom, .glyphicon-spin {
+    position: absolute;
+    top: 0;
+    left: 50%;
+    -webkit-animation: spin 1000ms infinite linear;
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: You don't need use prefix -webkit. It's put automatically with autoprefix. Remove this in all places, where is it

File: client/assets/less/pages/place-detail.less [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+    animation: spin 1000ms infinite linear;
+  }
+  @-webkit-keyframes spin {
```
Comment (max-yushchuk) [inline client/assets/less/pages/place-detail.less]: And remove it also

File: client/components/place/place-detail/map-place.factory.js [hunk 19]
```diff
@@ -0,0 +1,9 @@
+angular.module('mapModule')
+  .factory('mapPlaceFactory', ['constants', function(constants) {
+    /*var map =  L.map('map1', {
```
Comment (max-yushchuk) [inline client/components/place/place-detail/map-place.factory.js]: If you don't need it, please, remove this

**Comment by yurykozhenov** [comment]
Flexbox doesn't work on some photos

**Comment by yurykozhenov** [comment]
Fancybox doesn't work on some photos

**Comment by yurykozhenov** [comment]
Also, remove files associated with place modal window, since you don't use them anymore

**Pull request merged**